mundane-io/src/main/scala/com/ambiata/mundane/io/LocalDirectory.scala

```diff
@@ -156,33 +159,27 @@ class LocalDirectory private (val path: Path) extends AnyVal {
 }
 
 object LocalDirectory {
-  def Root: Path =
-    com.ambiata.mundane.path.Root
-
-  def Relative: Path =
-    com.ambiata.mundane.path.Relative
-
-  def fromFile(f: File): LocalDirectory =
-    unsafe(f.getPath)
-
-  def fromString(s: String): Option[LocalDirectory] =
-    s.split("/").toList match {
-      case Nil =>
-        new LocalDirectory(Root).some
-      case "" :: Nil =>
-        None
-      case "" :: parts =>
-        parts.traverse(Component.create).map(fromList(Root, _))
-      case parts =>
-        parts.traverse(Component.create).map(fromList(Relative, _))
-    }
+  def fromFile(f: File): RIO[LocalDirectory] =
+    LocalPath.fromFile(f).flatMap(_.determineDirectory)
+
+  def fromString(s: String): RIO[Option[LocalDirectory]] =
+    LocalPath.fromString(s).determinefWithPure(_ => none, _.some, none)
+
+  def fromList(dir: Path, parts: List[Component]): RIO[Option[LocalDirectory]] =
+    LocalPath.fromList(dir, parts).determinefWithPure(_ => none, _.some, none)
+
+  def fromURI(s: URI): RIO[Option[LocalDirectory]] =
+    LocalPath.fromURI(s).map(_.determineDirectory).sequence
```

Comment:
As discussed, wondering whether it's just as easy to have/leave these combinators on `LocalPath`, or whether it's nice to be able go directly to `LocalDirectory`. Vaguely related, it's interesting that half of these combinators return `Option` and half don't - is that intentional?